core/API/Request.php

```diff
@@ -329,11 +347,25 @@ public static function isRootRequestApiRequest()
      * @throws Exception
      */
     public static function isApiRequest($request)
+    {
+        $method = self::getMethodIfApiRequest($request);
+        return !empty($method);
+    }
+
+    /**
+     * Returns the current API method being executed, if the current request is an API request.
```

Comment:
I wonder if we need to clarify API request vs root API request... in this case here we would be also returning a method when an API request is being executed as part of a regular request... maybe we also need some wordings for it to better talk about it? not sure though.